src/sidebar/services/router.js

```diff
@@ -0,0 +1,85 @@
+import * as queryString from 'query-string';
+
+/**
+ * A service that manages the association between the route and route parameters
+ * implied by the URL and the corresponding route state in the store.
+ */
+// @ngInject
+export default function router($window, store) {
```

Comment:
This is using angular's `$window`, injected here. Do you foresee that we'll simply inject `window` later (using a non-ng injection mechanism)?

Reply:
Maybe. I think even in a post-Angular world it may still be useful to have a `$window` service to provide a test seam that is easier to mock than stubbing methods on the real `window` in tests, like some of our utilities have `window_` arguments.